runtime/compiler/optimizer/J9TransformUtil.cpp

```diff
@@ -1478,6 +1517,16 @@ J9::TransformUtil::transformIndirectLoadChainImpl(TR::Compilation *comp, TR::Nod
             if (value)
                {
                TR::SymbolReference *improvedSymRef = comp->getSymRefTab()->findOrCreateSymRefWithKnownObject(symRef, &value, isArrayWithConstantElements(symRef, comp));
+               int32_t stableArrayRank = isArrayWithStableElements(symRef, comp);
+
+               if (stableArrayRank > 0)
+                  {
+                  TR::KnownObjectTable *knot = comp->getOrCreateKnownObjectTable();
+                  if (knot)
+                     knot->addStableArray(&value, stableArrayRank);
+                  }
```

Comment:
If this were done within the conditional just below, it would respect the `performTransformation()`, and `addStableArray()` could take the known object index as a parameter instead of redundantly calling `getOrCreateIndexAt()`

Reply:
Improved.